### DISCUSSION TITLE
macOS arm64 build

### DISCUSSION THREAD

**Discussion by F9R** [WerWolv/ImHex#710] [Get Help] [answered]
Hi,
is there a arm64 build of ImHex for Apple Silicon m1? Software runs with Rosetta emulation but eg it crashes on quit. 
Regards

**Comment by WerWolv** [comment, marked as answer]
Hey
There aren't any Apple Silicon releases because I have no way of building them. You can compile it yourself through.
The crash on exit has been fixed in a later nightly, it happened on all OSes and architectures.

**Comment by iTrooz** [comment]
(If you build it from source, could you share the results here ? Just curious)

**Comment by F9R** [comment]
I set up my toolchain for building ImHex from git master but building failed. I think it failed because it requires Clang-15 for 'non-literal variables in constrexpr' compiler support. I got a lot of compiler-errors through build. Unfortunately I only found Clang-14 for apple silicon. Clang-15-rc3 was released just recently but it seems not available through MacPorts yet.

**Reply by WerWolv** [reply]
You want to use gcc for building. GCC 12 should work fine
